### PR TITLE
Gui testing take two

### PR DESCRIPTION
**Pull request by vigji**
PR replacing #113 after the latest changes, and thinking about GitHub Actions CI.

I would not keep this around for a lot of time. Currently it just features:
- some improvements on the setup.py;
- a version of sashimi GUI testing featuring for now just tab switching between modes.

My aim would be to  have tested also the experiment start/end in planar mode, ensuring that data are ok, then I would merge, if other people agree.

### CODE REVIEW

**Comment by diegoasua** [comment]
This is wonderful @vigji 🥇

**Comment by vigji** [comment]
Paradoxically, now `macos-latest` seems to be the testing OS where `pytest-qt` works in the most stable way ATM. I would leave only this OS in the testing matrix for now, as ultimately we just care about having all the interface parts in their place. Other reports of the issues I am having:

https://github.com/napari/napari/issues/804
https://github.com/pytest-dev/pytest-qt/issues/266

They are all quite recent, hope that a solution will emerge soon. Currently I don't find one for the windows problem, which is the only important one, as I would not explicitly support ubuntu, onto which we can't test too much.

**Comment by vigji** [comment]
A full run of a mock experiment in volumetric mode, plus checking the result, is now in place!

I would merge this ASAP, to have at least this basic testing in place for other PRs; we can add all the layers of complexity we want later!

File: tests/test_config.py [hunk 1]
```diff
@@ -19,6 +19,8 @@ def conf_path():
 
 def test_config_creation(conf_path):
     conf = config.read_config(file_path=conf_path)
+    print(conf_path)
+    print(conf)
```
Comment (vilim) [inline tests/test_config.py:23]: are these prints still necessary?

**Review by vilim — APPROVED** [review]
Looks great! Minor comments!

File: tests/test_config.py (hunk 1 above)
Comment (vigji) [reply tests/test_config.py]: No, removed in last push! I'll merge then

**Pull request merged**